packages/edit-site/src/components/dataviews/filter-enumeration.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as componentsPrivateApis } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { OPERATOR_IN } from './constants';
+
+const { DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem } = unlock(
+	componentsPrivateApis
+);
+
+export default function ( { filter, view, onChangeView } ) {
```

Comment:
I like that we now have a `FilterEnumeration` component. I imagine in the future we could have a `FilterString` or something else as well. I also like to keep all of these components with similar API (props). The current props look good to me.

What I don't like is that this component assumes that it's rendered within a dropdown menu. For me, this makes the component less reusable and it's also not something that we will be able to support for all types of filters (how do you put an input as a dropdown menu item). I'd have preferred if this component was just a normal form.

Reply:
> I'd have preferred if this component was just a normal form.

Could you clarify this bit? How would you do it?

I'd very much like having a standalone component that doesn't depend on the trigger component being any particular type. I'm just not sure how to achieve it with the existing component library. I've been resistant to open this PR until today, precisely because I didn't like this trigger/parent dependency.

Reply:
If I'm not wrong, dropdowns (not menu dropdowns) or modals can render any random component. So a design like something in my related comment here https://github.com/WordPress/gutenberg/pull/56479#issuecomment-1824499955 would be able to be achieved in a reusable component.

Pseudo code:

```
EnumarationFilter( props ) {

   return (
       <VStack>
             <Select label="operator" values={ [ 'in', 'not in' ] } />
             <Select label="value" multiple values={ props.filter.elements } />
       </VStack>
   );
}
```

Reply:
I see. Do you feel this PR would be useful towards implementing that vision (all components are centralized in a single place, so it's easier to change the UI for all of them at once)? Or do you feel we should clarify that vision first?

Reply:
I feel we should ship this one as it is and iterate on UI after. Everything (including updating UI) becomes easier.

Reply:
In my mind these intermediary steps are a detour and we should just build the generic filters. That said, I'm ok if we have different opinions here. So if you all agree about the current path, please go ahead. Nothing is set in stone.